src/wizard/wizard-step.component.js

```diff
@@ -0,0 +1,325 @@
+    } else {
+      ctrl.stepPriority = parseInt(ctrl.stepPriority);
+    }
+    if (angular.isUndefined(ctrl.okToNavAway)) {
+      ctrl.okToNavAway = true;
+    }
+    if (angular.isUndefined(ctrl.allowClickNav)) {
+      ctrl.allowClickNav = true;
+    }
+
+    ctrl.getEnabledSteps = function () {
+      return ctrl.steps.filter(function (step) {
+        return step.disabled !== 'true';
+      });
+    };
+
+    ctrl.getReviewSteps = function () {
+      var reviewSteps = ctrl.getEnabledSteps().filter(function (step) {
+        return !angular.isUndefined(step.reviewTemplate);
+      });
+      return reviewSteps;
+    };
+
+    ctrl.resetNav = function () {
+      ctrl.goTo(ctrl.getEnabledSteps()[0]);
+    };
+
+    ctrl.currentStepNumber = function () {
+      //retrieve current step number
+      return stepIdx(ctrl.selectedStep) + 1;
+    };
+
+    ctrl.getStepNumber = function (step) {
+      return stepIdx(step) + 1;
+    };
+
+    ctrl.isNextEnabled = function () {
+      var enabled = angular.isUndefined(ctrl.nextEnabled) || ctrl.nextEnabled;
+      if (ctrl.substeps) {
+        angular.forEach(ctrl.getEnabledSteps(), function (step) {
+          enabled = enabled && step.nextEnabled;
+        });
+      }
+      return enabled;
+    };
+
+    ctrl.isPrevEnabled = function () {
+      var enabled = angular.isUndefined(ctrl.prevEnabled) || ctrl.prevEnabled;
+      if (ctrl.substeps) {
+        angular.forEach(ctrl.getEnabledSteps(), function (step) {
+          enabled = enabled && step.prevEnabled;
+        });
+      }
+      return enabled;
+    };
+
+    ctrl.getStepDisplayNumber = function (step) {
+      return ctrl.pageNumber +  String.fromCharCode(65 + stepIdx(step)) + ".";
+    };
+
+    ctrl.prevStepsComplete = function (nextStep) {
+      var nextIdx = stepIdx(nextStep);
+      var complete = true;
+      angular.forEach(ctrl.getEnabledSteps(), function (step, stepIndex) {
+        if (stepIndex <  nextIdx) {
+          complete = complete && step.nextEnabled;
+        }
+      });
+      return complete;
+    };
+
+    ctrl.goTo = function (step) {
+      if (ctrl.wizard.isWizardDone() || !step.okToNavAway || step === ctrl.selectedStep) {
+        return;
+      }
+
+      if (firstRun || (ctrl.getStepNumber(step) < ctrl.currentStepNumber() && ctrl.selectedStep.prevEnabled) || ctrl.prevStepsComplete(step)) {
+        unselectAll();
+        ctrl.selectedStep = step;
+        if (step) {
+          step.selected = true;
+          ctrl.wizard.setPageSelected(step);
+
+          if (angular.isFunction (ctrl.selectedStep.onShow)) {
+            ctrl.selectedStep.onShow();
+          }
+
+          ctrl.currentStep = step.stepTitle;
+
+          firstRun = false;
+        }
+        ctrl.wizard.updateSubStepNumber (stepIdx(ctrl.selectedStep));
+      }
+    };
+
+    ctrl.stepClick = function (step) {
+      if (step.allowClickNav) {
+        ctrl.goTo(step);
+      }
+    };
+
+    this.addStep = function (step) {
```

Comment:
1 and 3 are identical, just a remnant of refactoring to not use scope anymore - I tried to clean those up as I went through but there were quite a few of them.  2 is internal and not exposed via the controller.